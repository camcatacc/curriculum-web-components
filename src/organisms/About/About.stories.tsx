import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import About from "organisms/About/About";
import AvatarImage from "stories/assets/avatar.jpg";
import Paragraph from "atoms/Paragraph/Paragraph";
import PageSize from "atoms/PageSize/PageSize";

const Template: ComponentStory<typeof About> = (args) => (
	<PageSize>
		<About {...args} />
	</PageSize>
);

export const Standard = Template.bind({});

Standard.args = {
	image: AvatarImage,
	name: "Pep",
	surname: "Guardiola",
	paragraphs: (
		<div className="flex flex-col gap-8">
			<Paragraph>
				I am a <b style={{ color: "#000243" }}>commited</b>, capable and resolutive person. I have skills, experience and formation as a{" "}
				<b style={{ color: "#000243" }}>developer</b>, which are very well complemented by my <b style={{ color: "#000243" }}>business background</b>.
			</Paragraph>
			<Paragraph>
				<p>
					I led an educational robotics and programming firm for five years. I devoted the last one and a half to ideate, design and develop an{" "}
					<b style={{ color: "#000243" }}>online solution in the form of a hybrid app</b> and a web platform.
				</p>
			</Paragraph>
			<Paragraph>
				<p>
					I am currently employed by Accenture as a <b style={{ color: "#545569" }}>full-stack developer</b>.
				</p>
			</Paragraph>
		</div>
	)
};

export default {
	title: "Organisms/About",
	component: About
} as ComponentMeta<typeof About>;
