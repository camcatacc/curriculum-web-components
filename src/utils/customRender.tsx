import { ThemeProvider } from "@material-ui/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { ReactElement } from "react";
import { getMaterialTheme } from "styles/materialUi";

export const customRender = (ui: ReactElement, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		return <ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>{children}</ThemeProvider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
