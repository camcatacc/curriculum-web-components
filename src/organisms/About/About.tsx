// Modules
import React from "react";
import { useTheme } from "@material-ui/core";

// Utils
import getMotionFlipOptions from "utils/getMotionFlipOptions";
import { getMotionFromOutsideVariants } from "utils/getMotionFromOutsideOptions";

// Elements
import AvatarWithName from "molecules/AvatarWithName/AvatarWithName";

// Styles
import * as Styled from "organisms/About/About.styled";

// Definitions

export interface AboutProps {
	name: string;
	surname: string;
	paragraphs: React.ReactNode;
	image: string;
	animated?: boolean;
}

// Functions
const getTextMotionOptions = (animated?: boolean) => {
	if (!animated) return {};
	const { initial, animation } = getMotionFromOutsideVariants("fromRight");
	return {
		variants: { initial, animation },
		initial: "initial",
		whileInView: "animation",
		viewport: {
			once: true,
			margin: "-200px"
		}
	};
};

// Element
const About = ({ name, surname, paragraphs, image, animated = false }: AboutProps) => {
	const textMotionOptions = getTextMotionOptions(animated);
	const imageMotionOptions = getMotionFlipOptions(animated, {});

	return (
		<Styled.ParentContainer className="flex flex-col md:flex-row items-center">
			<Styled.FirstChildContainer {...imageMotionOptions} className="w-3/5 md:w-auto">
				<AvatarWithName name={name} surname={surname} image={image} />
			</Styled.FirstChildContainer>
			<Styled.SecondChildContainer {...textMotionOptions} className="mt-4 md:mt-0 flex gap-8 flex-col justify-center">
				{paragraphs}
			</Styled.SecondChildContainer>
		</Styled.ParentContainer>
	);
};

export default About;
