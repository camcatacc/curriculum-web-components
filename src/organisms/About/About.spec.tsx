// Modules
import React from "react";
import { customRender } from "utils/customRender";

// Elements
import About from "organisms/About/About";

// Images
import Avatar from "assets/avatar.jpg";

// Definitions
import type { AboutProps } from "organisms/About/About";

// Default Props
const defaultProps: AboutProps = {
	name: "David",
	surname: "Camacho Cateura",
	paragraphs: <p>These are example paragraphs</p>,
	image: Avatar
};

// Tests
describe("About", () => {
	it("Renders as expected", () => {
		const { container } = customRender(<About {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});

	it("Renders as expected when animated", () => {
		const { container } = customRender(<About {...defaultProps} animated={true} />);
		expect(container).toMatchSnapshot();
	});
});
