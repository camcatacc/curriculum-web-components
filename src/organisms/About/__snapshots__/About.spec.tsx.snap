// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Renders as expected 1`] = `
<div>
  <div
    class="sc-dkPtRN rZzya flex flex-col md:flex-row items-center"
  >
    <div
      class="sc-hKwDye ffeJpd w-3/5 md:w-auto"
    >
      <div
        class="sc-gsDKAQ mtNEY w-full relative mb-4"
      >
        <img
          alt="DavidCamacho Cateura"
          class="absolute top-0 left-0 bottom-0 right-0 rounded-full"
          src="avatar.jpg"
        />
      </div>
      <h3
        class="MuiTypography-root sc-bdvvtL gjaeEB MuiTypography-h3 MuiTypography-alignCenter"
      >
        David
         
        Camacho Cateura
      </h3>
    </div>
    <div
      class="sc-eCImPb fwVAJe mt-4 md:mt-0 flex gap-8 flex-col justify-center"
    >
      <p>
        These are example paragraphs
      </p>
    </div>
  </div>
</div>
`;

exports[`About Renders as expected when animated 1`] = `
<div>
  <div
    class="sc-dkPtRN rZzya flex flex-col md:flex-row items-center"
  >
    <div
      class="sc-hKwDye ffeJpd w-3/5 md:w-auto"
      style="opacity: 0; transform: rotateY(90deg) translateZ(0);"
    >
      <div
        class="sc-gsDKAQ mtNEY w-full relative mb-4"
      >
        <img
          alt="DavidCamacho Cateura"
          class="absolute top-0 left-0 bottom-0 right-0 rounded-full"
          src="avatar.jpg"
        />
      </div>
      <h3
        class="MuiTypography-root sc-bdvvtL gjaeEB MuiTypography-h3 MuiTypography-alignCenter"
      >
        David
         
        Camacho Cateura
      </h3>
    </div>
    <div
      class="sc-eCImPb fwVAJe mt-4 md:mt-0 flex gap-8 flex-col justify-center"
      style="opacity: 0; transform: translateX(50px) translateY(0px) translateZ(0);"
    >
      <p>
        These are example paragraphs
      </p>
    </div>
  </div>
</div>
`;
